osf/models/user.py

```diff
@@ -680,14 +682,45 @@ def merge_user(self, user):
             node.save()
 
         # - projects where the user was the creator
-        user.created.filter(is_bookmark_collection=False).update(creator=self)
+        user.created.filter(is_bookmark_collection=False).exclude(type='osf.quickfiles').update(creator=self)
 
         # - file that the user has checked_out, import done here to prevent import error
         from osf.models import BaseFileNode
         for file_node in BaseFileNode.files_checked_out(user=user):
             file_node.checkout = self
             file_node.save()
 
+        # - move files in the merged user's quickfiles node, checking for name conflicts
+        from osf.models import QuickFiles
+        primary_quickfiles = QuickFiles.objects.get(creator=self)
+        merging_user_quickfiles = QuickFiles.objects.get(creator=user)
+
+        files_in_merging_user_quickfiles = BaseFileNode.objects.filter(node=merging_user_quickfiles, type='osf.osfstoragefile')
+        for merging_user_file in files_in_merging_user_quickfiles:
```

Comment:
The specific problem in the migration was only happening because of the pagination aspect -- without the pagination the queryset wasn't being re-evaluated every time.  I haven't caught a problem yet [with the merging user files tests I have](https://github.com/CenterForOpenScience/osf.io/pull/7339/files#diff-bfa630527bd745d1d1d5fe97eb1653d4R101) but let me know if there's a specific case this might be a problem for!